strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/maven/ChecksumMetadataExpirationStrategy.java

```diff
@@ -0,0 +1,83 @@
+package org.carlspring.strongbox.storage.metadata.maven;
+
+import org.carlspring.commons.encryption.EncryptionAlgorithmsEnum;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.repository.proxied.ProxyRepositoryArtifactResolver;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import static org.carlspring.strongbox.storage.metadata.maven.MetadataExpirationStrategy.Decision.*;
+
+@Component
+public class ChecksumMetadataExpirationStrategy
+        implements MetadataExpirationStrategy
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(ChecksumMetadataExpirationStrategy.class);
+
+    @Inject
+    private ProxyRepositoryArtifactResolver proxyRepositoryArtifactResolver;
+
+    public Decision decide(final RepositoryPath repositoryPath) throws IOException
+    {
+
+        RepositoryPath checksumRepositoryPath = resolveSiblingChecksum(repositoryPath, EncryptionAlgorithmsEnum.SHA1);
+        String currentChecksum = readChecksum(checksumRepositoryPath);
+        if (currentChecksum == null)
+        {
+            checksumRepositoryPath = resolveSiblingChecksum(repositoryPath, EncryptionAlgorithmsEnum.MD5);
+            currentChecksum = readChecksum(checksumRepositoryPath);
+            if (currentChecksum == null)
+            {
+                logger.debug("Unable to read local checksum for {}, will refetch metadata", repositoryPath.normalize());
+                return UNDECIDED;
+            }
+        }
+
+        proxyRepositoryArtifactResolver.fetchRemoteResource(checksumRepositoryPath);
+        final String newRemoteChecksum = readChecksum(checksumRepositoryPath);
+
+        if (newRemoteChecksum == null)
+        {
+            logger.debug("Unable to fetch remote checksum for {}, will refetch metadata", repositoryPath.normalize());
```

Comment:
log message could be more helpful if we knew which `EncryptionAlgorithmsEnum` is used